pkg/graveler/committed/apply_test.go

```diff
@@ -55,30 +52,94 @@ func TestApplyAdd(t *testing.T) {
 	}, summary)
 }
 
+func TestDiffRangesWithinSourceRange(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	source := testutil.NewFakeIterator()
+	source.
+		AddRange(&committed.Range{ID: "outer-range", MinKey: committed.Key("k1"), MaxKey: committed.Key("k6"), Count: 2}).
+		AddValueRecords(makeV("k1", "source:k1"), makeV("k6", "source:k6")).
+		AddRange(&committed.Range{ID: "last", MinKey: committed.Key("k10"), MaxKey: committed.Key("k13"), Count: 2}).
+		AddValueRecords(makeV("k10", "source:k10"), makeV("k13", "source:k13"))
+	diffs := testutil.NewFakeIterator()
+	diffs.
+		AddRange(&committed.Range{ID: "inner-range-1", MinKey: committed.Key("k2"), MaxKey: committed.Key("k3"), Count: 2}).
+		AddValueRecords(makeV("k2", "dest:k2"), makeV("k3", "dest:k3")).
+		AddRange(&committed.Range{ID: "inner-range-2", MinKey: committed.Key("k4"), MaxKey: committed.Key("k5"), Count: 2}).
+		AddValueRecords(makeV("k4", "dest:k4"), makeV("k5", "dest:k5"))
+	writer := mock.NewMockMetaRangeWriter(ctrl)
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k1", "source:k1")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k2", "dest:k2")))
+	writer.EXPECT().WriteRecord(gomock.Eq(*makeV("k3", "dest:k3")))
+	writer.EXPECT().WriteRange(gomock.Eq(*&committed.Range{ID: "inner-range-2", MinKey: committed.Key("k4"), MaxKey: committed.Key("k5"), Count: 2}))
```

Comment:
Not sure what `*&` does here, don't believe it is necessary.

```suggestion
	writer.EXPECT().WriteRange(gomock.Eq(committed.Range{ID: "inner-range-2", MinKey: committed.Key("k4"), MaxKey: committed.Key("k5"), Count: 2}))
```

Also below.